conjure_oxide/src/lib.rs

```diff
@@ -5,3 +5,46 @@ pub mod parse;
 pub use ast::Model;
 pub use error::Error;
 mod solvers;
+
+use versions::Versioning;
+
+pub fn conjure_executable() -> Result<String, String> {
+    let path = std::env::var("PATH")
+        .map_err(|_| "Could not read PATH environment variable".to_string())?;
+    let mut paths = std::env::split_paths(&path);
+    let conjure_dir = paths
+        .find(|path| path.join("conjure").exists())
+        .ok_or("Could not find conjure in PATH")?;
```

Comment:
I probably don't understand what this is doing, but normally you wouldn't expect to see the executable mentioned in the PATH, right? Instead it'll be stored in a dir that's in the path.

I think @ChrisJefferson wanted a way of being able to use a different conjure executable. We could do this via environment variables if we are keen to do it. CONJURE_BINARY="/path/to/custom/conjure-v3.1.4" etc, then we sould check whether that environment variable exists or not.

Reply:
I care abotu guarding against old/wrong conjure more than being able to provide a custom conjure at this point, FWIW. But no harm if we want to do both at the same time.

Reply:
I do think the first version doesn't need a way to set the paths -- it would be nice, but once this function exists it will be easier to add later.

Nice to see how conjure lets you work with the paths, I didn't know that, it's suprisingly nice!

Reply:
> I probably don't understand what this is doing, but normally you wouldn't expect to see the executable mentioned in the PATH, right? Instead it'll be stored in a dir that's in the path.
> 
> 
> 
> I think @ChrisJefferson wanted a way of being able to use a different conjure executable. We could do this via environment variables if we are keen to do it. CONJURE_BINARY="/path/to/custom/conjure-v3.1.4" etc, then we sould check whether that environment variable exists or not.

It's appending "conjure" onto each path in PATH and checking if the resulting (maybe) executable exists!

Reply:
Yes, I see now. Why not call conjure --version and see if it (a) throws an exception (b) happens to be the imagemagick one, our nemesis, or (c) an old version?

I guess you could produce a better error message if you knew our conjure was in the path but came after imagemagick, but we are not doing that either (unless I missed it).

Reply:
any thoughts on this @lixitrixi?

Reply:
> Yes, I see now. Why not call conjure --version and see if it (a) throws an exception (b) happens to be the imagemagick one, our nemesis, or (c) an old version?
> 
> 
> 
> I guess you could produce a better error message if you knew our conjure was in the path but came after imagemagick, but we are not doing that either (unless I missed it).

Oh haha this makes a lot more sense :) will do